core-bundle/src/EventListener/InsertTags/DateListener.php

```diff
@@ -0,0 +1,165 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\InsertTags;
+
+use Contao\Config;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\ServiceAnnotation\Hook;
+use Contao\Date;
+use Contao\PageModel;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Terminal42\ServiceAnnotationBundle\ServiceAnnotationInterface;
+
+/**
+ * Provides a {{format_date::<timestamp>::<format>}} and a
+ * {{convert_date::<date>::<source_format>::<target_format>}} Insert Tag.
+ *
+ * {{format_date::<timestamp>::<format>}}:
+ *
+ *   The first parameter is the date string to be parsed or a UNIX timestamp while
+ *   the second parameter is the date format. The third parameter can also be either
+ *   "date", "datim" or "time", which will use either the current page's or the
+ *   system's respective date format setting. If the third parameter is omitted,
+ *   "datim" will be used by default.
+ *
+ *   Usage:
+ *
+ *     {{format_date::2020-05-26 12:30:35::H:i, d.m.Y}}
+ *
+ *   Result:
+ *
+ *     12:30, 26.05.2020
+ *
+ * {{convert_date::<date>::<source_format>::<target_format>}}:
+ *
+ *   The first parameter is the date to be parsed and the second parameter the
+ *   format of that date. The third parameter defines the output date format.
+ *   The second and third parameter can also be "date", "datim" or "time", which
+ *   will use the current page's or the system's respective date format setting.
+ *
+ *   Usage:
+ *
+ *     {{convert_date::May 26th 2020, 12:30:35::F j\t\h Y, H:i:s::j. F Y, H:i:s}}
+ *
+ *   Result:
+ *
+ *     Tuesday, 26. May 2020, 12:30:35
+ *
+ * @internal
+ *
+ * @Hook("replaceInsertTags")
+ */
+class DateListener implements ServiceAnnotationInterface
+{
+    /**
+     * @var ContaoFramework
+     */
+    private $framework;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    public function __construct(ContaoFramework $framework, RequestStack $requestStack)
+    {
+        $this->framework = $framework;
+        $this->requestStack = $requestStack;
+    }
+
+    /**
+     * @return string|bool
+     */
+    public function __invoke(string $insertTag)
+    {
+        $tag = explode('::', $insertTag);
+
+        if (\in_array($tag[0], ['format_date', 'formatted_datetime'], true)) {
```

Comment:
I am not sure about handling the Haste insert tags here. We are basically hijacking the existing insert tags and taking them away from the original extension and I do not think that we should do such things.

Reply:
True. We can remove it, but I'll still do it if we have a consensus from the `codefog/contao-haste` maintainers :) /cc @qzminski @Toflar @aschempp

Reply:
I'm for removing it. Otherwise we would have to document it.